src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -741,9 +741,8 @@ int MPIDI_OFI_mpi_finalize_hook(void)
 
     /* Barrier over allreduce, but force non-immediate send */
     MPIDI_OFI_global.max_buffered_send = 0;
-    mpi_errno =
-        MPIR_Allreduce(&barrier[0], &barrier[1], 1, MPI_INT, MPI_SUM, MPIR_Process.comm_world,
-                       &errflag);
+    mpi_errno = MPIR_Allreduce_allcomm_auto(&barrier[0], &barrier[1], 1, MPI_INT, MPI_SUM,
```

Comment:
Does this override the user environment setting? Just trying to understand the scenario.

Reply:
I think the auto method just runs the default json selection, skipping user choice.